copier/main.py

```diff
@@ -177,82 +179,84 @@ def resolve_source_path(path):
+def resolve_paths(
+    src_path: StrOrPath, dst_path: StrOrPath, extra_paths: OptSeqStrOrPath
+) -> Tuple[Path, Path, OptSeqStrOrPath]:
     src_path = resolve_source_path(src_path)
     dst_path = Path(dst_path).resolve()
     extra_paths = [str(resolve_source_path(p)) for p in extra_paths or []]
     return src_path, dst_path, extra_paths
 
 
 def copy_local(
-    src_path,
-    dst_path,
-    data,
+    src_path: StrOrPath,
+    dst_path: StrOrPath,
+    data: AnyByStr,
     *,
-    extra_paths=None,
-    exclude=None,
-    include=None,
-    skip_if_exists=None,
-    tasks=None,
-    envops=None,
-    **flags
-):
+    extra_paths: OptSeqStrOrPath = None,
+    exclude: OptSeqStrOrPath = None,
+    include: OptSeqStrOrPath = None,
+    skip_if_exists: OptSeqStrOrPath = None,
+    tasks: OptSeqStr = None,
+    envops: Optional[AnyByStr] = None,
+    **flags: bool
+) -> None:
     src_path, dst_path, extra_paths = resolve_paths(src_path, dst_path, extra_paths)
     config_data = load_config_data(src_path, quiet=flags["quiet"])
-    user_exclude = config_data.pop("_exclude", None)
+    user_exclude: List[str] = config_data.pop("_exclude", None)
     if exclude is None:
         exclude = user_exclude or DEFAULT_EXCLUDE
 
-    user_include = config_data.pop("_include", None)
+    user_include: List[str] = config_data.pop("_include", None)
     if include is None:
         include = user_include or DEFAULT_INCLUDE
 
-    user_skip_if_exists = config_data.pop("_skip_if_exists", None)
+    user_skip_if_exists: List[str] = config_data.pop("_skip_if_exists", None)
     if skip_if_exists is None:
         skip_if_exists = user_skip_if_exists or []
 
-    user_tasks = config_data.pop("_tasks", None)
+    user_tasks: List[str] = config_data.pop("_tasks", None)
     if tasks is None:
         tasks = user_tasks or []
 
-    user_extra_paths = config_data.pop("_extra_paths", None)
+    user_extra_paths: List[str] = config_data.pop("_extra_paths", None)
     if not extra_paths:
         extra_paths = [str(resolve_source_path(p)) for p in user_extra_paths or []]
 
     user_data = config_data if flags["force"] else query_user_data(config_data)
     data.update(user_data)
     data.setdefault("folder_name", dst_path.name)
-    render = get_jinja_renderer(src_path, data, extra_paths, envops)
+    engine = get_jinja_renderer(src_path, data, extra_paths, envops)
 
-    skip_if_exists = [render.string(pattern) for pattern in skip_if_exists]
+    skip_if_exists = [engine.string(pattern) for pattern in skip_if_exists]
     must_filter, must_skip = get_name_filters(exclude, include, skip_if_exists)
 
     if not flags["quiet"]:
         print("")  # padding space
 
-    for folder, _, files in os.walk(str(src_path)):
-        rel_folder = folder.replace(str(src_path), "", 1).lstrip(os.path.sep)
-        rel_folder = render.string(rel_folder)
-        rel_folder = rel_folder.replace("." + os.path.sep, ".", 1)
+    for _folder, _, files in os.walk(str(src_path)):
```

Comment:
Trying out `typing.cast()` here, instead of defining new variables.

Reply:
Prevented the introduction of new variables by moving type declaration before the for loop.